plugins/repository-s3/src/main/java/org/elasticsearch/cloud/aws/InternalAwsS3Service.java

```diff
@@ -221,4 +210,22 @@ protected void doClose() throws ElasticsearchException {
         // Ensure that IdleConnectionReaper is shutdown
         IdleConnectionReaper.shutdown();
     }
+
+    static class PrivilegedInstanceProfileCredentialsProvider implements AWSCredentialsProvider {
+        private final InstanceProfileCredentialsProvider credentials;
+
+        private PrivilegedInstanceProfileCredentialsProvider(InstanceProfileCredentialsProvider credentials) {
+            this.credentials = credentials;
```

Comment:
You could just make this construct the InstanceProfileCredentialsProvider here, and have no ctor args.